server/src/MessageHandler.ts

```diff
@@ -5,77 +5,190 @@ import {
   Range,
   Location,
   DeclarationParams,
+  CompletionParams,
+  CompletionList,
+  CompletionItem,
+  Position,
 } from 'vscode-languageserver'
 import * as util from './util'
 import { fullDocumentRange } from './provider'
-import { getDMMF } from '@prisma/sdk'
-import { TextDocument, Position } from 'vscode-languageserver-textdocument'
+import { TextDocument } from 'vscode-languageserver-textdocument'
 import format from './format'
+import {
+  getSuggestionsForAttributes,
+  getSuggestionsForTypes,
+  getSuggestionForBlockTypes,
+  getSuggestionForFirstInsideBlock,
+  getSuggestionForSupportedFields,
+  getSuggestionsForInsideAttributes,
+} from './completions'
 
-function getWordAtPosition(document: TextDocument, position: Position): string {
-  const currentLine = document.getText({
-    start: { line: position.line, character: 0 },
-    end: { line: position.line, character: 9999 },
+function getCurrentLine(document: TextDocument, line: number) {
+  return document.getText({
+    start: { line: line, character: 0 },
+    end: { line: line, character: 9999 },
   })
+}
+
+function convertDocumentTextToTrimmedLineArray(
+  document: TextDocument,
+): Array<string> {
+  return [...Array(document.lineCount)].map((_, i) =>
```

Comment:
Any `[...array]` operation is very expensive. I suggest using `.fill()` instead, it's on my machine about 15-20 times faster. See benchmark:
https://jsbench.me/kckackxd2n/1
```suggestion
  return Array(document.lineCount).fill().map((_, i) =>
```

Reply:
`fill()` expects a value though - I set that to 0